src/main/java/com/google/api/codegen/common/OutputProvider.java

```diff
@@ -12,19 +12,19 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.google.api.codegen.gapic;
+package com.google.api.codegen.common;
 
-import com.google.api.codegen.GeneratedResult;
 import java.io.IOException;
 import java.util.Collection;
 import java.util.Map;
+
 /**
- * A GapicProvider performs file generation (code, static resources, etc.) using a proto-based Model
- * for a particular language.
+ * An OutputProvider performs file generation (code, config, etc) or file copying (static resources)
+ * based on input provided at construction time.
  *
- * @param <T> Class which represent generated file body (may be {@code byte[]})
+ * @param <T> Class which represents the generated file body.
  */
-public interface GapicProvider<T> {
+public interface OutputProvider<T> {
```

Comment:
`OutputProvider` is a very generic name. And it doesn't seem to really be a "provider" - it is a generator. `OutputFileGenerator` may be better, or `OutputGenerator` (but that is even more generic).

Reply:
It seems like the right name for this is `Generator`; I think the reason we didn't do this before is that there is already an inner interface of `SnippetSetRunner` named `Generator`. It looks like that is only used by client config generation though; all other generation goes through `CommonSnippetSetRunner`. Maybe we should rename that inner interface so we can use the name. I think `Generator` is appropriate for the use case here because the output is `GeneratedResult`, and the interface really is quite general. What do you think? I would also be ok with `OutputGenerator` as you suggest, although the `Output` part is potentially so generic that it is redundant. We could potentially go for `CodeGenerator` also, with the understanding that config, readmes, etc all count as code. I'm a little wary of `FileGenerator` because no files are actually generated at this stage.

Reply:
`CodeGenerator` or `Generator` SGTM.